packages/client/lib/sync/fetcher/fetcher.ts

```diff
@@ -264,7 +264,7 @@ export abstract class Fetcher<JobTask, JobResult, StorageItem> extends Readable
       try {
         await this.store(result)
         this.finished++
-        this.emit('fetched', result)
+        this.config.events.emit(Event.SYNC_FETCHER_FETCHED, result as any)
```

Comment:
Just had another short look here, do we now always have to cast arguments on `emit` as `any` and the typing benefits are just on the listening side or is this some special case here? 🤔

Reply:
This one is a special case since `result` is of type `StorageItem[]` that is  `Block[] | BlockHeader[]`.

We can see if we can improve this local code to better infer `result` as `Block[] | BlockHeader[]` before emitting it, then we wouldn't need the `any`.

Reply:
Ok, but generally we can go ahead here with this PR, right?